### ISSUE TITLE
[0.13.7-RC3] "test->test" does not work when cached resolution enabled

### ISSUE THREAD

**Issue by xuwei-k** [sbt/sbt#1716]
https://github.com/xuwei-k/sbt-0.13.7-RC3-bug
### Step

`b/src/test/scala/Test.scala`

``` scalaa
import org.scalatest.FunSpec
```

`project/build.properties`

```
sbt.version=0.13.7-RC3
```

`build.sbt`

``` scala
val commonSettings = Seq(
  scalaVersion := "2.11.4",
  updateOptions := updateOptions.value.withCachedResolution(true)
)

lazy val a = project.settings(commonSettings: _*).settings(
  libraryDependencies += "org.scalatest" %% "scalatest" % "2.2.2" % "test"
)

lazy val b = project.settings(commonSettings: _*).dependsOn(
  a,
  a % "test->test"
)
```

and

`sbt test:compile`
### actual

```
[info]   Compilation completed in 14.753 s
[error] /home/travis/build/xuwei-k/sbt-0.13.7-RC3-bug/failure/b/src/test/scala/Test.scala:1: object scalatest is not a member of package org
[error] import org.scalatest.FunSpec
[error]            ^
[error] one error found
[error] (b/test:compile) Compilation failed
[error] Total time: 23 s, completed Nov 6, 2014 7:57:44 PM
```
### expect

`sbt test:compile` should be success

**Comment by eed3si9n** [comment]
The fix I made for #1711 wasn't correct.

**Comment by xuwei-k** [comment]
I confirmed that correction with v0.13.7-RC4. thanks!
